### DISCUSSION TITLE
OSET License, formatted with Markdown

### DISCUSSION THREAD

**Discussion by stratofax** [ion-oset/developer-commons#10] [Infrastructure]
Late last year, I reformatted the OSET License using Markdown and used it as the [BallotLab/LICENSE.md at main · TrustTheVote-Project/BallotLab](https://github.com/TrustTheVote-Project/BallotLab/blob/main/LICENSE.md). The correct vertsion number for this license document is "2.1.1" and has been approved by both John Sebes and Greg Miller. 

I'd like to propose **using this formatted verion of the OSET License as the official OSET License file**, and encourage its use across all OSET repos. 

This seems like a great test case for adding a file to the main branch of this repo. As I understand, the process would look something like this:

1. Present the file to the group as a discussion (this post)
2. Solicit feedback in the discussion (also, this)
3. Create an Issue describe the intent and requirements and how the proposed changes meet them.
4. Create a branch of the repo to track the changes to the license file
5. Update the file as needed, based on the feedback.
6. Create a Pull Request for the branch with the proposed license file updates
5. Link to the new PR to solicit final approval (or to implement any required changes)
6. Close the Issue by approving the PR and merging the branch to main. 

I've attached the proposed LICENSE.md as a file to this discussion. This file is a snapshot of the file linked above and shouldn't be edited, but ultimately incorporated into a branch on this repo.

I'd like to hear your comments, feedback and suggestions for both the file itself and the proposed approval process.

[LICENSE.md](https://github.com/ion-oset/developer-commons/files/8090741/LICENSE.md)

## To Do

These tasks are for @stratofax unless otherwise noted.

- [ ] Find canonical, orginal text version of LICENSE.txt in an existing TTV repo.
- [ ] Create a `license` branch in this repo to track changes
- [ ] Add the original LICENSE.txt file to the `license` branch
- [ ] Rename 'LICENSE.txt` to `LICENSE.md` [using GitHub tools to rename](https://docs.github.com/en/repositories/working-with-files/managing-files/renaming-a-file) so we can see a diff of the changes, instead of deleting one file and creating another.
- [ ] Incorporate the proposed changes to `LICENCE.md` in a commit to the 'license` branch.
- [ ] Create a pull request for the team to review and attach it to the corresponding Issue.
- [ ] Merge the pull request (also close the Issue and delete the branch).

**Comment by ion-oset** [comment]
On the approval process:

- I think 4 and 5 should be swapped. It's preferable to create a issue with the formal proposal, and have that reviewed before the work starts on the PR. As changes are made it can be updated. This is partly because I think we should be [drawing a line between informal work and formal proposals](https://github.com/ion-oset/developer-commons/discussions/4#discussioncomment-2207300) and I want to see a formal review process tracked in an issue, and proposals have a lifecycle that doesn't necessarily result in acceptance. It's also because GitHub's workflow supports the flow of Discussion -> Issue -> PR. 

- Editing a file over and over again is going to get annoying to track changes on. I suggest it's preferrable to use either Wiki pages or Gists so we can see actual diffs in the license text.

On the license itself:

- Do you know if older versions of the license are available? It might make sense to have more of the history of them. (Though obviously we should check on that.)
- The link to the license online is using `osetfoundation.org` instead of `osetinstitute.org`. We can wait until a revised version appears and update accordingly, but we might also want to flag that.

**Reply by stratofax** [reply]
Thanks for the comments. My replies by section:

On the approval process:

- I agree that it's better to create the Issue first (swap 4 & 5), then go back and link the PR to the original issue. It's easier to create an Issue than a PR, and writing up  the issue will help organize how the PR is structured.
- I don't think we should edit a file here in Discussions. I included the file so we had a snapshot of what's in the linked BallotLab repo. I think the best way to handle the edits is to create a branch off of main and make the edits there, then add a PR and link back to the (yet to be created) Issue.

On the license itself:

- There is an older, text only version of the license (v.2.1) and I agree it would make sense to use that as the starting point of the branch history.
- I agree, let's update osetfoundation.org to osetinstitute.org in the final version of the updated license.

**Reply by stratofax** [reply]
Also, I updated the original discussion post with the changes described in Ion's comments and my replies.

**Reply by ion-oset** [reply]
> I think the best way to handle the edits is to create a branch off of main and make the edits there

That works. I had been thinking "PR" where you said "branch", which would require an issue. That works for a formal proposal, but not an informal one. But we can compare [across branches](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/viewing-and-comparing-commits/comparing-commits#comparing-branches) or [across forks](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/viewing-and-comparing-commits/comparing-commits#comparing-across-forks) without a PR or an issue.

**Comment by stratofax** [comment]
I added a "To Do" checklist to the OP on this discussion. Checkbox formatting is a little fincky, but works great when you get it right!

## Checkbox Checklist

- [ ] Start with a dash (`-`)
- [ ] Add a space (`- `)
- [ ] Add an open bracket, space and close bracket (`- [ ]`)
- [ ] Add another space and type your checkbox item (`- [ ] Sample item text`)
- [ ] Press Return and GitHub will create another checkbox for you automatically
- [x] Add an `x` to check a checkbox (`- [x] Checked item`)
- [x] You have to add the `x` manually, though.
- [x] Checking a checkbox in HTML (formatted) view updates the underlying Markdown text -- no need to open the editor!
